src/main/resources/META-INF/plugin.xml

```diff
@@ -188,67 +188,68 @@
         <additionalTextAttributes scheme="Default" file="colorSchemes/SonarLintDefault.xml"/>
         <additionalTextAttributes scheme="Darcula" file="colorSchemes/SonarLintDarcula.xml"/>
         <editorActionHandler action="EditorEscape" implementationClass="org.sonarlint.intellij.editor.EscapeHandler"/>
-        <projectService serviceImplementation="org.sonarlint.intellij.editor.EditorDecorator"/>
-        <projectService serviceImplementation="org.sonarlint.intellij.actions.SonarLintToolWindow"/>
         <errorHandler implementation="org.sonarlint.intellij.errorsubmitter.BlameSonarSource"/>
         <iw.actionProvider implementation="org.sonarlint.intellij.editor.SonarLintEditorWidgetActionProvider"/>
 
         <!-- Empty language attribute corresponds to any language (see com.intellij.lang.Language.ANY) -->
         <externalAnnotator language="" implementationClass="org.sonarlint.intellij.editor.SonarExternalAnnotator"/>
 
+        <!-- Cannot be moved to service annotation because of 'serviceInterface' / 'testServiceImplementation' -->
         <applicationService serviceInterface="org.sonarlint.intellij.core.server.events.ServerEventsService"
                             serviceImplementation="org.sonarlint.intellij.core.server.events.ServerEventsProductionService"
                             testServiceImplementation="org.sonarlint.intellij.util.ServerEventsNoOpService"/>
+
+        <!-- Cannot be moved to service annotation because of persisting application state -->
         <applicationService serviceImplementation="org.sonarlint.intellij.config.global.SonarLintGlobalSettingsStore"/>
+
+        <!-- Cannot be moved to service annotation because of 'serviceInterface' / 'testServiceImplementation' -->
         <applicationService serviceInterface="org.sonarlint.intellij.core.EngineManager"
                             serviceImplementation="org.sonarlint.intellij.core.DefaultEngineManager"
                             testServiceImplementation="org.sonarlint.intellij.core.TestEngineManager"/>
+
         <applicationService serviceImplementation="org.sonarlint.intellij.SonarLintPlugin"/>
+
+        <!-- Cannot be moved to service annotation because of 'serviceInterface' / 'testServiceImplementation' -->
         <applicationService serviceInterface="org.sonarlint.intellij.util.GlobalLogOutput"
                             serviceImplementation="org.sonarlint.intellij.util.GlobalLogOutputImpl"
                             testServiceImplementation="org.sonarlint.intellij.util.GlobalLogOutputTestImpl"/>
+
         <applicationService serviceImplementation="org.sonarlint.intellij.util.SonarLintActions"/>
+
+        <!-- Cannot be moved to service annotation because of 'serviceInterface' -->
         <applicationService serviceInterface="org.sonarlint.intellij.telemetry.SonarLintTelemetry"
```

Comment:
I might be wrong but for service classes that implement an interface it might still work, you can probably annotate the class that inherits the interface, but still use the interface when getting the service

Reply:
I'll try that on Monday.

Reply:
Investigated that: All the ones accesing the service via the interface cannot be migrated as of today. The docs are quite sparse on that but I found some info nonetheless. Maybe they can be migrated in the future, looks like they try to simplify it even further.